docs/source/en/tasks/object_detection.md

```diff
@@ -394,211 +1441,118 @@ Hugging Face Hub. Upon training completion, push the final model to the Hub as w
- Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.681
- Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.292
- Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.168
- Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.208
- Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.429
- Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.274
- Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.484
- Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.501
- Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.191
- Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.323
- Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.590
+>>> # This one is for visualization with no padding
+>>> resize_only = A.Compose([
+...     A.LongestMaxSize(max_size=max_size),
+... ])
 ```
-These results can be further improved by adjusting the hyperparameters in [`~transformers.TrainingArguments`]. Give it a go!
-
-## Inference
-Now that you have finetuned a DETR model, evaluated it, and uploaded it to the Hugging Face Hub, you can use it for inference.
-The simplest way to try out your finetuned model for inference is to use it in a [`Pipeline`]. Instantiate a pipeline
-for object detection with your model, and pass an image to it:
 
+Load model and image processor from the Hugging Face Hub (skip to use already trained in this session):
 ```py
->>> from transformers import pipeline
->>> import requests
-
->>> url = "https://i.imgur.com/2lnWoly.jpg"
->>> image = Image.open(requests.get(url, stream=True).raw)
-
->>> obj_detector = pipeline("object-detection", model="devonho/detr-resnet-50_finetuned_cppe5")
->>> obj_detector(image)
+>>> model_repo = "qubvel/detr_finetuned_cppe5"
+>>> image_processor = AutoImageProcessor.from_pretrained(model_repo)
+>>> model = AutoModelForObjectDetection.from_pretrained(model_repo)
 ```
 
-You can also manually replicate the results of the pipeline if you'd like:
+And detect bounding boxes:
 
 ```py
->>> image_processor = AutoImageProcessor.from_pretrained("devonho/detr-resnet-50_finetuned_cppe5")
->>> model = AutoModelForObjectDetection.from_pretrained("devonho/detr-resnet-50_finetuned_cppe5")
+>>> np_preprocessed_image = resize_and_pad(image=np.array(image))["image"]
 
 >>> with torch.no_grad():
-...     inputs = image_processor(images=image, return_tensors="pt")
-...     outputs = model(**inputs)
-...     target_sizes = torch.tensor([image.size[::-1]])
-...     results = image_processor.post_process_object_detection(outputs, threshold=0.5, target_sizes=target_sizes)[0]
+...     inputs = image_processor(images=[np_preprocessed_image], return_tensors="pt")
+...     outputs = model(inputs["pixel_values"].cuda())
```

Comment:
nit - here we're assuming there's a cuda device available and the model has been loaded onto it. I'd make it mode explicit by defining a `device` var which we then do `.to(device)` for both the inputs and the model. This way, it's a simple one-line change for someone to run on e.g. cpu or xpu.

Reply:
good point!